contracts/Core/StakeManager.sol

```diff
@@ -183,30 +183,27 @@ contract StakeManager is Initializable, ACL, StakeStorage, StateManager, Pause {
         require(sToken.balanceOf(msg.sender) >= lock.amount, "locked amount lost"); // Can Use ResetLock
 
         uint256 rAmount = _convertSRZRToRZR(lock.amount, staker.stake, sToken.totalSupply());
-        require(sToken.burn(msg.sender, lock.amount), "Token burn Failed");
+        uint256 lockAmount = lock.amount;
         staker.stake = staker.stake - rAmount;
-
         // Function to Reset the lock
         _resetLock(stakerId);
-
+        require(sToken.burn(msg.sender, lockAmount), "Token burn Failed");
```

Comment:
Hey cant we place this before _resetLock(), hence we won't need to store lockAmount, locally 
_resetLock is an internal function so there is no reentrancy possibility.

Reply:
yes we can do that

Reply:
fixed